### ISSUE TITLE
[Discussion] Apps do not update during core upgrade process if they are not enable.

### ISSUE THREAD

**Issue by davitol** [owncloud/market#422]
## Steps to reproduce

1 . Having an app installed  in our server but not enabled. (Disabled in ```occ app:list```section)
2. run ```occ upgrade``` in order to upgrade the server

## Expected result

Should disabled apps get updated too or not?

@VicDeo @PVince81 @pmaier1 @IljaN

**Comment by VicDeo** [comment]
imho:  disabled = unused so no action is needed.

**Comment by IljaN** [comment]
> imho: disabled = unused so no action is needed

What happens if the core gets updated past the min/max constraint of the disabled app? If admin suddenly decides he wants to use this app he can not re-enable it for a update?

**Comment by pmaier1** [comment]
Intuitively I would want disabled apps to be upgraded as well (e.g. security fixes) as they are still installed, just not in use. When the admin activates them again (which is the purpose of disabling an app) it would be good to have the current version (currently admins probably won't even notice when this is not the case). When admins don't want to have apps anymore, they should uninstall them.

Opinion @PVince81 ?

**Comment by PVince81** [comment]
yeah, I'd also prefer that as it's more intuitive than expecting admins to remember to update the app with "occ market:upgrade" first. if we were to show a warning, not sure where and how.
I don't think we'd want to have "occ app:enable X" to a marketplace check and say "oh by the way, the app you enabled is outdated, you should consider updating" as it's already too late.

**Comment by cdamken** [comment]
We should add a flag in the case that you don't want to upgrade the app while enabling it:

example
```
occ app:enable bookmarks --do-not-upgrade
```
in case that is the version you already tested in your staging system
